rest/RESTClient.ts

```diff
@@ -66,7 +67,16 @@ export default class RESTClient {
         const discordBucket = api.headersIn['x-ratelimit-bucket'] as string | undefined;
         if (api.headersIn['x-ratelimit-global'] === 'true') {
           this.globallyRateLimited = true;
-          setTimeout(() => { this.globallyRateLimited = false; }, api.json.retry_after * 1e3);
+          this.queue.push({ method, endpoint: endpointFinal, auth, payload });
+
+          setTimeout(async () => {
+            this.globallyRateLimited = false;
+
+            const nextRequest = this.queue[0];
+            await this.request(nextRequest.method, nextRequest.endpoint, nextRequest.auth, nextRequest.payload);
+
+            this.queue.filter((req) => req !== nextRequest);
```

Comment:
This only removes the first item in the queue, not any others. I'd recommend putting a while loop here checking the array length, and shifting the first element out and using that to request. This also means you can get rid of the unnecessary filter in the last line here